### DISCUSSION TITLE
autoconvert '.' to '->' when the type is a pointer

### DISCUSSION THREAD

**Discussion by WiiliamC** [microsoft/vscode-cpptools#9598] [Q&A]
According to [#862](https://github.com/microsoft/vscode-cpptools/issues/862), the autoconversion has been completed. Yet I have v1.10.8 but I can't find how to enable the autoconversion.
When I type ".", there is no autocompletion list:
![image](https://user-images.githubusercontent.com/48458974/179573360-079f6206-ec80-4074-aab2-4b1a2197503f.png)
When I type "->":
![image](https://user-images.githubusercontent.com/48458974/179573236-f83d06b3-8ee3-4efb-8a56-bb640cc73813.png)
I've gone through the C/C++ settings but haven't found any related items. The "IntelliSenseEngine" is set to "default".
Any help would be appreciated!

**Comment by sean-mcmanus** [comment]
It works for me. Can you provide more repro details? What is the type of root?

```cpp
class CC
{
public:
    int b;
};

int main()
{
    CC *cc;
    cc.; // completes .b to ->b
    return 0;
}
```

**Reply by WiiliamC** [reply]
I found that only pointer in struct/class cannot be auto completed:
![GIF 2022-7-19 4-12-33](https://user-images.githubusercontent.com/48458974/179609381-ffa848eb-4d9c-449a-bed2-03614c223231.gif)
Here is the code:
```cpp
struct Point
{
    int x, y;
    Point(int a, int b) : x(a), y(b) {}
};

struct Line
{
    Point *p0, *p1;
    void print()
    {
        p0. // no auto-completion
    }
};

void printPoint()
{
    Point *p0, *p1;
    p0->x // auto-completed
}
```

**Reply by sean-mcmanus** [reply]
I've filed a bug at https://github.com/microsoft/vscode-cpptools/issues/9599

**Reply by sean-mcmanus** [reply]
FYI, a workaround is to make an edit in the method definition, such as via adding a //, moving the cursor to a new line, and then removing the //.
![image](https://user-images.githubusercontent.com/19859882/179852922-b769ed7e-f84a-487d-9be7-d6ca34933287.png)

**Reply by sean-mcmanus** [reply]
@WiiliamC Fixed with 1.12.0 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.12.0
